modules/couchbase/src/main/java/org/testcontainers/couchbase/CouchbaseContainer.java

```diff
@@ -15,447 +15,423 @@
+    private static final int KV_PORT = 11210;
 
-    @Getter
-    private boolean query = true;
+    private static final int KV_SSL_PORT = 11207;
 
-    @Getter
-    private boolean index = true;
+    private static final String DOCKER_IMAGE_NAME = "couchbase/server";
 
-    @Getter
-    private boolean primaryIndex = true;
+    private static final String VERSION = "6.5.0";
 
-    @Getter
-    private boolean fts = false;
+    private static final ObjectMapper MAPPER = new ObjectMapper();
 
-    @Getter(lazy = true)
-    private final CouchbaseEnvironment couchbaseEnvironment = createCouchbaseEnvironment();
+    private static final OkHttpClient HTTP_CLIENT = new OkHttpClient()
+        .newBuilder()
+        .readTimeout(Duration.ofMinutes(1))
+        .build();
 
-    @Getter(lazy = true)
-    private final CouchbaseCluster couchbaseCluster = createCouchbaseCluster();
+    private String username = "Administrator";
 
-    private List<BucketAndUserSettings> newBuckets = new ArrayList<>();
+    private String password = "password";
 
-    private String urlBase;
+    private Set<CouchbaseService> enabledServices = EnumSet.allOf(CouchbaseService.class);
 
-    private SocatContainer proxy;
+    private final List<BucketDefinition> buckets = new ArrayList<>();
 
+    /**
+     * Creates a new couchbase container with the default image and version.
+     */
     public CouchbaseContainer() {
-        this(DOCKER_IMAGE_NAME + VERSION);
+        this(DOCKER_IMAGE_NAME + ":" + VERSION);
     }
 
-    public CouchbaseContainer(String imageName) {
+    /**
+     * Creates a new couchbase container with a custom image name.
+     *
+     * @param imageName the image name that should be used.
+     */
+    public CouchbaseContainer(final String imageName) {
         super(imageName);
-
-        withNetwork(Network.SHARED);
-        setWaitStrategy(new HttpWaitStrategy().forPath("/ui/index.html"));
     }
 
-    @Override
-    public Set<Integer> getLivenessCheckPortNumbers() {
-        return Sets.newHashSet(getMappedPort(REST));
+    /**
+     * Set custom username and password for the admin user.
+     *
+     * @param username the admin username to use.
+     * @param password the password for the admin user.
+     * @return this {@link CouchbaseContainer} for chaining purposes.
+     */
+    public CouchbaseContainer withCredentials(final String username, final String password) {
+        checkNotRunning();
+        this.username = username;
+        this.password = password;
+        return this;
     }
 
-    @Override
-    protected void configure() {
-        if (clusterPassword.length() < REQUIRED_DEFAULT_PASSWORD_LENGTH) {
-            logger().warn("The provided cluster admin password length is less then the default password policy length. " +
-                "Cluster start will fail if configured password requirements are not met.");
-        }
+    public CouchbaseContainer withBucket(final BucketDefinition bucketDefinition) {
```

Comment:
Just a general thing with our APIs, sometimes `with` will be a plain setter, sometimes add something to a collection. This depends on the context, but might not always be obvious to the user. Sometimes we also use `add` to indicate this.

Reply:
We have a plenty of `withX` that `add` (`withExposedPort`, `withCreateContainerCmdModifier` and others), I would keep it as is, especially given that I expect most of the users to only need one bucket anyways